c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/AgentManifest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.api;
+
+import org.apache.nifi.c2.protocol.component.api.BuildInfo;
+import org.apache.nifi.c2.protocol.component.api.Bundle;
+import org.apache.nifi.c2.protocol.component.api.ComponentManifest;
+import org.apache.nifi.c2.protocol.component.api.SchedulingDefaults;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+@ApiModel
+public class AgentManifest implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private String identifier;
+    private String agentType;
+    private String version;
+    private BuildInfo buildInfo;
+    private List<Bundle> bundles;
+    private ComponentManifest componentManifest;
```

Comment:
@arpadboda is this top level `AgentManifest.componentManifest` used by nifi-minifi-cpp today? If not (ie, all components exist under a Bundle), we could remove it from this API module

Reply:
I see no top level ComponentManifest in the minifi c++ AgentManifest code. The relevant code is [here](https://github.com/apache/nifi-minifi-cpp/blob/main/libminifi/include/core/state/nodes/AgentInformation.h#L614), but it may not be an easy read.

Reply:
@szaszm Looking through that code, I agree, this field is not used by the c++ agent. As we have no need/plan to use it in the Java agent, I would recommend removing it for now @mattyb149